### PR TITLE
[PLAYER-2911] SSAI Web Heartbeat

### PR DESCRIPTION
**Pull request by cacel**
Heartbeat for Ooyala Players & Non-Ooyala Playeres

### CODE REVIEW

File: WebSampleApps/WebHeartbeatSSAISampleApp/ssai_heartbeat.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    }
+
+    function _onPlayheadTimeChange() {
+        movieDuration = video.duration || 0;
+        playheadPosition = video.currentTime || 0;
+    }
+
+    function startHeartBeat() {
+        stopHeartBeat();
+
```
Comment (dallen42) [inline WebSampleApps/WebHeartbeatSSAISampleApp/ssai_heartbeat.js]: @cacel could probably remove this empty line

File: WebSampleApps/WebHeartbeatSSAISampleApp/ssai_heartbeat_ooyala.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+      log('initialization completed', config);
+    }
+
+    function stopHeartBeat() {
+      clearInterval(heartbeatTimer);
+    }
+
+    function startHeartBeat() {
+      stopHeartBeat();
+
```
Comment (dallen42) [inline WebSampleApps/WebHeartbeatSSAISampleApp/ssai_heartbeat_ooyala.js]: @cacel could probably remove this empty line

File: WebSampleApps/WebHeartbeatSSAISampleApp/ssai_heartbeat_ooyala.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    function parseGUID(url) {
+      var reg = new RegExp(/ssai_guid=([^&?]*)/g);
+      var result = reg.exec(url);
+
+      if ( result.length && result[1] ) {
+        return result[1];
+      }
+
+      return '';
+
```
Comment (dallen42) [inline WebSampleApps/WebHeartbeatSSAISampleApp/ssai_heartbeat_ooyala.js]: @cacel don't need this extra line

**Review by dallen42 — APPROVED** [review]
@cacel LGTM apart from a few formatting suggestions

**Comment by cacel** [comment]
@dallen42 I made the changes you requested.

**Review by ipadilla4 — APPROVED** [review]
LGTM

File: WebSampleApps/WebHeartbeatSSAISampleApp/ssai_heartbeat.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    var DEFAULT_CONFIG = Object.freeze({
+        Interval: 10 * 1000,
+        ReportingPathPattern: '<hostname>/v1/vod_playback_pos/<embed_code>?ssai_guid=<ssai_guid>'
+    });
+    
+    var baseURL = '//ssai.ooyala.com/vhls/ltZ3l5YjE6lUAvBdflvcDQ-zti8q8Urd/RpOWUyOq86gFq-STNqpgzhzIcXHV/eyJvIjoiaHR0cDovL3BsYXllci5vb3lhbGEuY29tL2hscy9wbGF5ZXIvYWxsL2x0WjNsNVlqRTZsVUF2QmRmbHZjRFEtenRpOHE4VXJkLm0zdTg_dGFyZ2V0Qml0cmF0ZT0yMDAwJnNlY3VyZV9pb3NfdG9rZW49YTFkQmFraElRa3hzTDBSak9GbFBZVWd4TURWS01XUjVhMGwwWkhRNFVuUlRRM1p2VTNONVFXbDRkSEpQTW5GUFFVVkRSR3REUWtsUVIwZEpDbFkyVFdOelpHOUdaSGhtU0cxWVkzaGtORWQ1Ym1JM2VGbDNQVDBLIiwiZSI6IjE1MjQ2NzcwMTYiLCJzIjoiTThYdzhGckM5MFhEdnZlZmxWMVZxT0FQOE9TSThZRWI1dDJGSTRxWElQVT0ifQ==/manifest.m3u8?ssai_guid=HeartbeatSampleTest';
+    var embed_code = 'ltZ3l5YjE6lUAvBdflvcDQ-zti8q8Urd';
+    var streamURL = '';
+    var ssai_guid = '';
```
Comment (pilievOoyala) [inline WebSampleApps/WebHeartbeatSSAISampleApp/ssai_heartbeat.js:13]: I know we merged this already but I noticed this in another checkin. Can you please make all the variable camelCase and not use snakecase if possible?

**Pull request merged**